src/main/java/gui/MainWindow.java

```diff
@@ -42,6 +56,104 @@ public MainWindow(String title) {
+        menuBar.add(layersMenu);
+
+        setJMenuBar(menuBar);
+    }
+
+    private void loadCollectionsToListModel() {
+
+        for (int i = 0; i < nodeCollection.getArrayList().size(); i++) {
+            nodesListModel.add(i, nodeCollection.getArrayList().get(i));
+        }
+
+        for (int i = 0; i < edgeCollection.getArrayList().size(); i++) {
+            edgesListModel.add(i, edgeCollection.getArrayList().get(i));
+        }
+
+        for (int i = 0; i < locationCollection.getArrayList().size(); i++) {
+            locationsListModel.add(i, locationCollection.getArrayList().get(i));
+        }
+
+    }
+
+    private void loadJsonsFromFolder(String folderPath) {
+
+        String edgePath = folderPath + ResourceBundle.getBundle("strings").getString("json_edge");
+        String nodePath = folderPath + ResourceBundle.getBundle("strings").getString("json_node");
+        String floorPath = folderPath + ResourceBundle.getBundle("strings").getString("json_floor");
+        String locationPath = folderPath + ResourceBundle.getBundle("strings").getString("json_location");
+        String tagPath = folderPath + ResourceBundle.getBundle("strings").getString("json_tag");
+
+        try {
+            ArrayList<Edge> edges = JsonParser.getEntityArrayList(edgePath, new TypeToken<List<Edge>>() {
+            }.getType());
+            edgeCollection = (EdgeArray) addArrayToCollection(edges, edgeCollection);
+
+            ArrayList<Node> nodes = JsonParser.getEntityArrayList(nodePath, new TypeToken<List<Node>>() {
+            }.getType());
+            nodeCollection = (NodeArray) addArrayToCollection(nodes, nodeCollection);
+
+            ArrayList<Floor> floors = JsonParser.getEntityArrayList(floorPath, new TypeToken<List<Floor>>() {
+            }.getType());
+            floorCollection = (FloorArray) addArrayToCollection(floors, floorCollection);
+
+            ArrayList<Location> locations = JsonParser.getEntityArrayList(locationPath, new TypeToken<List<Location>>() {
+            }.getType());
+            locationCollection = (LocationArray) addArrayToCollection(locations, locationCollection);
+
+            ArrayList<Location_Tag> location_tags = JsonParser.getEntityArrayList(tagPath, new TypeToken<List<Location_Tag>>() {
+            }.getType());
+            locationTagCollection = (Location_TagArray) addArrayToCollection(location_tags, locationTagCollection);
+
+            JOptionPane.showMessageDialog(getParent(), ResourceBundle.getBundle("strings").getString("json_success"));
+
+        } catch (FileNotFoundException e) {
+
+            JOptionPane.showMessageDialog(getParent(),
+                    ResourceBundle.getBundle("strings").getString("json_error"),
+                    ResourceBundle.getBundle("strings").getString("error"),
+                    JOptionPane.ERROR_MESSAGE);
+        }
+
+    }
+
+    private <T> HashSet<?> addArrayToCollection(ArrayList<T> entityArrayList, HashSet<Object> entityCollection) {
```

Comment:
Ponieważ ArrayListy oraz HashSet rozszerzają interfejs Collection, to metoda _addAll_ hash setu przyjmuje array listę bez problemu. Całą tą metodą można śmiało zamienić takim wywołaniem:
`entityCollection.addAll(entityArrayList);`

Reply:
Piotr wyjaśnił mi, że wywołanie metody add w pętli jest podyktowane tym, że metoda add jest rozszerzona, a addAll już nie. W takiej sytuacji poleciłbym rozszerzyć również tą metodę. Nie mniej jednak, całość rozwiązania z rozszerzaniem hash setu uważam z błędne. Lepszym rozwiązaniem byłoby owrapować kolekcję, zamiast ją rozszerzać.